### PR TITLE
#11 Added support for .NET Core

### PR DESCRIPTION
**Pull request by niemyjski**
@apawsey, @maxhauser, @herecydev I've converted the project to the new project format and used  compiler directives so the full framework existing code bases wasn't touched except for the ctor overload for version will throw an argument null exception if version is null.

- [x] I noticed the tests are failing on .NET Core, I was hoping you guys could take a look at this... I think it has to do with the equals implementation.
- [x] We need to setup a build server. I can use my account on appveyor and get the build all going. We need to update the build system to do assembly info patching + replace the version in project.json. If we use appveyor I have these scripts already built.
- [x] @maxhauser I only have build targets for .net standard and .net 4.5.2. I can add back the old targets very easily if you want.. I just bumped it to 4.5.2 as that's the oldest version that's supported by microsoft.

### CODE REVIEW

**Comment by maxhauser** [comment]
- I will take a look at the failing tests.
- It would be cool if you could setup the build server.
- It is ok for me to remove support for old .net versions - but then the major version should be incremented because of the breaking change.

**Comment by maxhauser** [comment]
I removed the string interning - the performance penalty should be minimal if visible at all and I could clean up the code by removing some of the conditional directives.
you know - premature optimization...
All tests are green again.

File: src/Semver/SemVersion.cs [hunk 1]
```diff
@@ -129,21 +110,13 @@ public static SemVersion Parse(string version, bool strict = false)
             if (!match.Success)
                 throw new ArgumentException("Invalid version.", "version");
 
-#if NETSTANDARD
```
Comment (niemyjski) [inline src/Semver/SemVersion.cs]: @maxhauser I think you want to revert this change. I'm a bit worried about this in all of my projects because I really don't know how parsing foreign formats will work in .net standard and there has been limited docs / feedback on this.

**Comment by niemyjski** [comment]
I'll work on the build server / nuget package creation sometime this week. I'm currently traveling and on limited connectivity.

File: src/Semver/SemVersion.cs (hunk 1 above)
Comment (maxhauser) [inline src/Semver/SemVersion.cs]: @niemyjski you are probably right. Although I cannot think of language dependent differences in integer parsing - there may be some.

**Comment by herecydev** [comment]
Code looks good to me, the ifdefs are upsetting but unavoidable. Would be good to track these as "tech debt" and look to bring the functionality back if/when it becomes available, e.g. Compiled regex.

**Comment by niemyjski** [comment]
@maxhauser I added the appveyor script. Can you create a new appveyor account (if you don't have one) and just link it to this github repo and then it should be good to go. You'll only need to fill out the appveyor notification settings as well as github token (they are encrypted via the appveyor). Then the build should be all set.

**Comment by maxhauser** [comment]
@niemyjski done - looking good.

**Comment by niemyjski** [comment]
Might want to update the read me with a badge of the build status

**Comment by niemyjski** [comment]
I think everything is done, the only thing that needs to be done is updating the appveyor script file to push to nuget (you might need your credentials?) and then merge this branch :)
